nodes/image_processing.py

```diff
@@ -316,6 +317,146 @@ def execute(
         )
 
 
+class ResizeWithMegapixels:
+    """Resizes images and masks to a target megapixel count while preserving aspect ratio.
+
+    A specialized resizing node that targets a specific image size in megapixels rather than
+    exact dimensions. This is useful for batch processing images to a consistent size while
+    maintaining their original proportions and managing memory usage.
+
+    Args:
+        megapixels (float): Target size in megapixels (0.01-100.0)
+        image (torch.Tensor, optional): Input image in BWHC format with values in range [0, 1]
+        mask (torch.Tensor, optional): Input mask in BWHC format with values in range [0, 1]
+        interpolation (str): Resampling method:
+            - "bilinear": Linear interpolation (smooth)
+            - "nearest": Nearest neighbor (sharp)
+            - "bicubic": Cubic interpolation (smoother)
+            - "area": Area averaging (good for downscaling)
+        antialias (bool): Whether to apply antialiasing when downscaling
+
+    Returns:
+        tuple:
+            - image (torch.Tensor): Resized image in BWHC format
+            - mask (torch.Tensor): Resized mask in BWHC format
+
+    Raises:
+        ValueError: If neither image nor mask is provided
+        ValueError: If megapixels is outside valid range
```

Comment:
Is this true? I could not find that in the code and when I tried it the UI will just set the maximum.

Reply:
Since it was on the resize node, and I'm calling it on this node is probably why it was added.